modApiServer/test/org/aion/api/server/pb/ApiAion0Tests.java

```diff
@@ -0,0 +1,2137 @@
+import org.apache.commons.lang3.RandomUtils;
+import org.junit.Test;
+
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Collections;
+
+import static org.junit.Assert.*;
+
+public class ApiAion0Tests {
+
+    private byte[] msg, socketId, hash;
+
+    private static final int MSG_HASH_LEN = 8;
+    private static final int RSP_HEADER_NOHASH_LEN = 3;
+    private static final int REQ_HEADER_NOHASH_LEN = 4;
+    private static final int RSP_HEADER_LEN = RSP_HEADER_NOHASH_LEN + MSG_HASH_LEN;
+
+    public ApiAion0Tests() {
+        msg = "test message".getBytes();
+        socketId = RandomUtils.nextBytes(5);
+        hash = RandomUtils.nextBytes(ApiUtil.HASH_LEN);
+    }
+
+    private byte[] stripHeader(byte[] rsp) {
+        boolean hasHash = (rsp[2] == 1);
+        int bodyLen = rsp.length - (hasHash ? RSP_HEADER_LEN : RSP_HEADER_NOHASH_LEN);
+
+        if (hasHash) {
+            return Arrays.copyOfRange(rsp, RSP_HEADER_LEN, RSP_HEADER_LEN + bodyLen);
+        } else {
+            return Arrays.copyOfRange(rsp, RSP_HEADER_NOHASH_LEN, RSP_HEADER_NOHASH_LEN + bodyLen);
+        }
+    }
+
+    @Test
+    public void TestCreate() {
+        System.out.println("run TestCreate.");
+        AionImpl impl = AionImpl.inst();
+        ApiAion0 api = new ApiAion0(impl);
+        api.shutDown();
+    }
+
+    @Test
+    public void TestHeartBeatMsg() {
+        System.out.println("run TestHeartBeatMsg.");
+        AionImpl impl = AionImpl.inst();
+        ApiAion0 api = new ApiAion0(impl);
+
+        byte[] msg = ByteBuffer.allocate(api.getApiHeaderLen()).put(api.getApiVersion())
+                .put((byte) Message.Servs.s_hb_VALUE).array();
+        assertTrue(ApiAion0.heartBeatMsg(msg));
+        assertFalse(ApiAion0.heartBeatMsg(null));
+
+        msg[0] = 0;
+        assertFalse(ApiAion0.heartBeatMsg(msg));
+
+        api.shutDown();
+    }
+
+    @Test
+    public void TestProcessProtocolVersion() {
+        System.out.println("run TestProcessProtocolVersion.");
+        AionImpl impl = AionImpl.inst();
+        ApiAion0 api = new ApiAion0(impl);
+
+        byte[] request = ByteBuffer.allocate(msg.length + REQ_HEADER_NOHASH_LEN + hash.length).put(api.getApiVersion())
+                .put((byte) Message.Servs.s_net_VALUE)
+                .put((byte) Message.Funcs.f_protocolVersion_VALUE)
+                .put((byte) 1)
+                .put(hash)
+                .put(msg).array();
+
+        byte[] rsp = api.process(request, socketId);
+
+        assertEquals(Message.Retcode.r_success_VALUE, rsp[1]);
+
+        try {
+            Message.rsp_protocolVersion pv = Message.rsp_protocolVersion.parseFrom(stripHeader(rsp));
+            assertEquals(Integer.toString(api.getApiVersion()), pv.getApi());
+            assertEquals(Version.KERNEL_VERSION, pv.getKernel());
+            assertEquals(EquihashMiner.VERSION, pv.getMiner());
+        }
+        catch (Exception e) {
```

Comment:
very minor - typical java style places the `catch` on the same line as the `}`
(also `fail()` can print a message so you could put the `e.getMessage()` inside the `fail` :) )

I don't think these `try-catch` blocks are too necessary though since what they do is print a fail message and then fail. If you don't use these blocks the exception will be thrown and cause the test to fail anyway - plus we will get a stack trace instead of just the error message - which is more handy for debugging

Reply:
+1 for abandoning the catch block in favor or just saying 'throws Exception' in method signature

Reply:
done